src/fragments/Graph.react.js

```diff
@@ -226,8 +226,11 @@ class PlotlyGraph extends Component {
         if (!layout) {
             return layout;
         }
-
-        return mergeDeepRight(layout, this.getLayoutOverride(responsive));
+        let override = this.getLayoutOverride(responsive);
+        for (let key in override) {
+            layout[key] = override[key];
+        }
+        return layout;  // not really a clone
```

Comment:
Something like this? I'm not sure how to keep original state though. If we e.g. store `autosize` here, we cannot simply restore it the next time, because we don't know whether the user changed `autoresize` on the figure. But I may be misunderstanding.

Reply:
Yes, this looks good for applying the overrides. Then for updates, I guess the logic would go something like:
- We can see if the layout is being changed by looking at what's recorded on the plot div when we're about to replot: `const layoutChanged = gd.layout !== props.layout.figure`
- If `layoutChanged`, store these attributes in state and then apply the overrides.
- Otherwise reapply the original values from state and then apply the overrides (which should give the right result regardless of whether `responsive` changed or not.